### PR TITLE
Add Id to ConnectionInfo

### LINKED ISSUE #827: Consider exposing IHttpConnectionFeature.ConnectionId through ConnectionInfo
What it says on the tin. Is there a reason this isn't exposed through the `HttpContext` surface?

### PR DESCRIPTION
**Pull request by khellang**
Take it or leave it :wink: I went with `TraceIdentifier` to match the request ID on `HttpContext`.

Closes #827

### CODE REVIEW

File: src/Microsoft.AspNetCore.Http/Internal/DefaultConnectionInfo.cs [hunk 1]
```diff
@@ -20,25 +20,21 @@ public class DefaultConnectionInfo : ConnectionInfo
 
         public DefaultConnectionInfo(IFeatureCollection features)
         {
-            Initialize(features);
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http/Internal/DefaultConnectionInfo.cs]: Seems unrelated.

Comment (khellang) [inline src/Microsoft.AspNetCore.Http/Internal/DefaultConnectionInfo.cs]: Yes, it is. I thought it made sense to clean it up when touching the file anyway. I can strip the last commit if you want...

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http/Internal/DefaultConnectionInfo.cs]: Yep. Do that.

Comment (khellang) [inline src/Microsoft.AspNetCore.Http/Internal/DefaultConnectionInfo.cs]: Done! 👍

File: src/Microsoft.AspNetCore.Http.Abstractions/ConnectionInfo.cs [hunk 2]
```diff
@@ -10,6 +10,11 @@ namespace Microsoft.AspNetCore.Http
 {
     public abstract class ConnectionInfo
     {
+        /// <summary>
+        /// Gets or sets a unique identifier to represent this connection in trace logs.
+        /// </summary>
+        public abstract string TraceIdentifier { get; set; }
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Http.Abstractions/ConnectionInfo.cs]: :-1: this is mixing names. We already have a TraceIdentifier on HttpContext with a different source value (actually made for tracing). Just call it ConnectionId

Comment (khellang) [inline src/Microsoft.AspNetCore.Http.Abstractions/ConnectionInfo.cs]: So... `HttpContext.Connection.ConnectionId`? Smurf much? 😉

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Http.Abstractions/ConnectionInfo.cs]: Or HttpContext.Connection.Id?

Comment (khellang) [inline src/Microsoft.AspNetCore.Http.Abstractions/ConnectionInfo.cs]: Sure, that was next on my list after `TraceIdentifier` 😄

**Comment by khellang** [comment]
:heart:
